### PR TITLE
Test me0 muon reco 76 x v1

### PR DESCRIPTION
**Pull request by pietverwilligen**
Dear Reco Conveners, dear Tracking POG Conveners

This PR contains a few changes in the tracking code to allow Run-1 tracking in absence of the calorimeter geometries. For this I had to hide some calorimeter calls behind TrackAssociatorParameters and I made a separate configuration file for Run-1 vertexing.
Please review and let me know your comments.
greets
Piet

### CODE REVIEW

**Comment by slava77** [comment]
Hi Piet,
I was able to run standard reco (almost out of the box) + ME0 code as a part of #11398 
please check https://github.com/cms-sw/cmssw/pull/11398#issuecomment-146037225

**Comment by pietverwilligen** [comment]
Hi Slava
I was talking to @dnash86  and he told me that for him this suggestion was not working. I tried out myself your suggestion  3d07963 and also for me it does not work:

----- Begin Fatal Exception 08-Oct-2015 23:28:30 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 1 event: 1
   [1] Running path 'reconstruction_step'
   [2] Calling event method for module BunchSpacingProducer/'bunchSpacingProducer'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: int
Looking for module label: addPileupInfo
Looking for productInstanceName: bunchSpacing

You like probably to look at the Time module:

%MSG
TimeModule> 1 1 gtEvmDigis L1GlobalTriggerEvmRawToDigi 0.000415087
TimeModule> 1 1 siPixelDigis SiPixelRawToDigi 0.00491595
TimeModule> 1 1 siStripDigis SiStripRawToDigiModule 0.0117681
TimeModule> 1 1 ecalDigis EcalRawToDigi 0.052644
TimeModule> 1 1 ecalPreshowerDigis ESRawToDigi 0.00903082
TimeModule> 1 1 hcalDigis HcalRawToDigi 0.238525
TimeModule> 1 1 muonCSCDigis CSCDCCUnpacker 0.0322399
TimeModule> 1 1 muonDTDigis DTUnpackingModule 0.026294
TimeModule> 1 1 muonRPCDigis RPCUnpackingModule 0.00756884
TimeModule> 1 1 castorDigis CastorRawToDigi 0.0486431
TimeModule> 1 1 scalersRawToDigi ScalersRawToDigi 0.00400305
TimeModule> 1 1 tcdsDigis TcdsRawToDigi 0.000179052
TimeModule> 1 1 bunchSpacingProducer BunchSpacingProducer 0.0225799
----- Begin Fatal Exception 08-Oct-2015 23:28:30 CEST-----------------------

I was using the wantSummary of process.Options:

TrigReport ---------- Modules in Path: reconstruction_step ------------
TrigReport  Trig Bit#    Visited     Passed     Failed      Error Name
TrigReport     1    0          1          0          0          1 bunchSpacingProducer
TrigReport     1    1          0          0          0          0 siPixelClustersPreSplitting
TrigReport     1    2          0          0          0          0 siPixelRecHitsPreSplitting

Now, looking at this (I have no idea what exactly this bunchSpacingProducer is) it might be that you are able to fix this problem. But what I see, and what worries me, is that you are running the Run-2 reconstruction. Now what I learned is that the run-2 iterative tracking relies on the calorimeters for some "PreSplitting" work.  The configuration file I created --- which I agree will be impossible to maintain, but is useful for us at this time, and we won't use it in the future --- runs the Run-1 iterative tracking, which does not rely on the calorimeters and is exactly what we need in this situation.
greets
Piet

**Comment by slava77** [comment]
Hi Piet,

The bunchspacing made from the pileupinfo doesn't have much to do with reconstruction.
It can be fixed up.

Tracking doesn't depend on calorimeter inputs in a way that it wouldn't run: if calorimeter is empty, the run2 iterative tracking will run just fine (the jet core iteration will not be triggered, but you probably don't care about it ... and you don't have it in the run1 tracking setup you are trying )

**Comment by pietverwilligen** [comment]
mmmhmm ... it is not that the calorimeter is empty ... the problem is that a call to any calorimeter geometry results in a crash (null pointer).

**Comment by slava77** [comment]
On 10/8/15 5:09 PM, Piet Verwilligen wrote:
> mmmhmm ... it is not that the calorimeter is empty ... the problem is
> that a call to any calorimeter geometry results in a crash (null pointer).
> 

in my comments, I provided a reco config file, which works.
Please check and see if it runs for you and makes sensible results
(it ran for me on 100  single-muon events).

**Comment by pietverwilligen** [comment]
I tried this file:
https://github.com/cms-sw/cmssw/commit/3d07963f89007867be59970f6a94ce9442a1ac7f#diff-f33bdba7c6b5d51b46b54ca79c5b9fba
do you mean you have another file?

**Comment by pietverwilligen** [comment]
I am running in CMSSW_7_6_X_2015-10-07-2300, which has integrated  #11398 
Is there something that I miss?

**Comment by slava77** [comment]
On 10/8/15 5:20 PM, Piet Verwilligen wrote:
> I tried this file:
> 3d07963#diff-f33bdba7c6b5d51b46b54ca79c5b9fba
> <https://github.com/cms-sw/cmssw/commit/3d07963f89007867be59970f6a94ce9442a1ac7f#diff-f33bdba7c6b5d51b46b54ca79c5b9fba>
> do you mean you have another file?

that's the one I used.
Apply the changes in that commit and then

SingleMuPt100_cfi_GEM-SIM_Extended2015MuonGEMDev_RPCGEMME0Customs_cfg.py
SingleMuPt100_cfi_DIGI_Extended2015MuonGEMDev_RPCGEMME0Customs_cfg.py
runTestUpgradeGEMME0Reco_cfg.py


> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/11700#issuecomment-146702562>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 TAMU: Physics Dept Texas A&M MS4242, College Station, TX 77843-4242
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------

**Comment by pietverwilligen** [comment]
Well thats exactly what i did and for me it was not working with the error message reported earlier.
To be explicit, I ran exactly those GEN-SIM and DIGI step and I ran your modified version of runTestUpgradeGEMME0Reco_cfg.py.

**Comment by slava77** [comment]
On 10/8/15 5:35 PM, Piet Verwilligen wrote:
> Well thats exactly what i did and for me it was not working with the
> error message reported earlier.
> To be explicit, I ran exactly those GEN-SIM and DIGI step and I ran your
> modified version of runTestUpgradeGEMME0Reco_cfg.py.
> 

do you have a copy on lxplus to make a diff?

my copy runs; maybe I missed something when committing

**Comment by slava77** [comment]
looking closer, your error makes sense: #11576  is another consumer of bunchSpacing from addPileupInfo and it was merged after my test setup.
I recovered it in the DIGI file
https://github.com/slava77/cmssw/compare/0bfa10d...slava77:CMSSW_7_6_0_pre6/sign604/pull11398-fixRunTest

**Comment by pietverwilligen** [comment]
Hi Slava

Thanks. That fix for the digi file is working. I ll update this request such that it contains these changes.

**Comment by pietverwilligen** [comment]
Hi Slava

I tried to translate your lines in the DIGI config file to the gemcustoms such that the cmsDriver command works out of the box. 

Now, the DIGI step worked fine and indeed the bunchSpacing is kept in the event content. But when running your simplified reconstruction configuration, I get a crash ... Can you have a look at the gemcustoms and tell me where I went wrong?
Thanks
Piet

**Comment by slava77** [comment]
Piet,
What's the error in the reco step?

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: RecoLocalMuon/GEMRecHit/test/runTestUpgradeGEMME0Reco_v2_cfg.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (slava77) [inline RecoLocalMuon/GEMRecHit/test/runTestUpgradeGEMME0Reco_v2_cfg.py]: I don't think that we need to proliferate configurations serving the same purpose.
Considering that RecoLocalMuon/GEMRecHit/test/runTestUpgradeGEMME0Reco_cfg.py is not maintainable in its current form, I suggest to add all relevant changes from this file there and remove this _v2

**Comment by slava77** [comment]
@pietverwilligen 
please follow up on the previous comments or suggest other steps.
Thank you.

**Comment by slava77** [comment]
-1
needs follow up
I propose to continue in 80X, unless significant need is present to have this in 760

**Pull request closed without merging**